### DISCUSSION TITLE
help on sentiment analysis: limiting rows by sentiment value

### DISCUSSION THREAD

**Discussion by jshinnn** [winkjs/wink-nlp#115] [Help on coding] [answered]
Hi, i'm developing a data visualization and want to do something similar to [this](https://observablehq.com/@winkjs/how-to-perform-sentiment-analysis). However, my text is too large so there are a lot of rows in the table. is there a way to limit the number of rows in a table, or sort the rows by ascending / descending sentiment value, or filter out what shows up in the rows based on if it's above a certain sentiment value? Thanks!

**Comment by sanjayaksaxena** [comment]
Hello @jshinnn 

There could be several ways this can be achieved depending on the context and volume of the data. Here is a simple way to only show rows with sentiment > threshold value (`0.2` in the example):

<img width="858" alt="image" src="https://user-images.githubusercontent.com/29990/224897253-cb06f245-bd34-438b-ac0b-f6dd261dd28f.png">

Best,
Sanjaya

**Reply by jshinnn** [reply]
hey Sanjaya, thanks for your thoughtful reply! this worked table-wise but unfortunately i see the text that is used to build the rows of the table presented right before the row. see below: any tips?
<img width="909" alt="Screen Shot 2023-03-14 at 1 18 03 AM" src="https://user-images.githubusercontent.com/72616591/224938386-0b4b81a4-9bad-47d6-805e-05d633208054.png">

**Reply by sanjayaksaxena** [reply]
Hi @jshinnn 

Seems strange! The code contained in the Observable notebook and also the code suggested above should not exhibit this behaviour.

Can you share the code that is being used in your notebook?

Best,
Sanjaya

**Reply by jshinnn** [reply]
hi @sanjayaksaxena , i tried it on some smaller text with the code you suggested and am having the same problem. the link to the observable notebook is [here](https://observablehq.com/d/38a600b18c8e150e). thank you!
<img width="889" alt="Screen Shot 2023-03-14 at 11 59 00 AM" src="https://user-images.githubusercontent.com/72616591/225109560-a651da00-f2b6-45b1-87a3-16fd4e6ddf6e.png">

**Reply by sanjayaksaxena** [reply]
My bad @jshinnn! Here is the working code with a threshold `0.45`:

<img width="849" alt="image" src="https://user-images.githubusercontent.com/29990/225216476-497abd45-eb0e-4a33-aa5f-12fff6209a90.png">

**Reply by jshinnn** [reply]
@sanjayaksaxena thank you so much! i appreciate it. as i am analyzing a large dataset, being able to filter helps a lot but i was wondering if there's a way to make the table scrollable in a fixed frame? otherwise, the table is very large as there are many sentences in my dataset. please let me know if this doesnt make sense.

**Comment by sanjayaksaxena** [comment, marked as answer]
Consider using [Observable Table](https://observablehq.com/@observablehq/input-table)... It will require you to make some changes in the code:

<img width="1300" alt="image" src="https://user-images.githubusercontent.com/29990/225271500-136e60e2-79bf-476d-b7fb-099b26ffa60d.png">

**Reply by jshinnn** [reply]
@sanjayaksaxena this is great, thanks so much! one last question, do you know how to make it so that the text in the "sentence" column shows all the words? for example, some sentences are too long so they end in ellipses. i tried looking on the [observable table input page ](https://observablehq.com/@observablehq/input-table) directly but can't seem to find a solution.  
<img width="848" alt="Screen Shot 2023-03-15 at 1 17 39 PM" src="https://user-images.githubusercontent.com/72616591/225432224-cfedb54a-97f6-46ae-873d-f05fd9bd7c86.png">

**Reply by rachnachakraborty** [reply]
Hey @jshinnn

Setting the layout to auto should display the entire content of a cell. 

Citing from [observable table input page](https://observablehq.com/@observablehq/input-table)

```
There’s a similar _width_ option if you want to give certain columns specific
widths. The table component defaults to a fixed _layout_ if there are twelve or
fewer columns; this improves performance and avoids reflow when scrolling.

You can switch _layout_ to auto if you prefer sizing columns based on content.
This makes the columns widths resize with the data, which can cause the columns
to jump around as the user scrolls. A horizontal scroll bar is added if the
total width exceeds the table’s width.

```

Hope this helps.

Best,
Rachna

**Reply by jshinnn** [reply]
hello @rachnachakraborty thank you so much! i was wondering if it's possible to display the full text but in rows, instead of horizontal scrollbar as the `auto` layout produces.
